cuda_phase-retrieval_class/src/phase_retrieval.cu

```diff
@@ -4,6 +4,7 @@
 
 #include "cupy_ref.hpp"
 #include "cupy_caster.hpp"
+#include "cuda_algo.hpp"
```

Comment:
The `#include` should be done in the files, where the kernel are used means:
* cuda_phase-retrieval_class/src/phase_algo.hpp 
* cuda_phase-retrieval_class/test/test_6_cupy_reference_management/test_algo.hpp 

At the moment, you have a strange dependency. Your actual algorithm is only compiling, if you create python bindings of it.

Reply:
yes, right now I am confused about where should I put the includes properly because it always repeating in almost every files

Reply:
Yes, managing the includes is a hard job. Unfortunately, I don't know a good way, to do this in generic way. Most of the time, I view the headers and check if they are related. And for clean up the includes in the beginning of the file, I simply comment out the headers and check, if it still compiling.